DeAceroDLabs/marketplace-frontend
# [DMP-496] Display Products on cart view & [DMP-498] Add delete functionality

## What is done

[[DMP-496] Publicar nombre de usuario en Auth0](https://deacero.atlassian.net/browse/DMP-496)
[[DMP-498] Publicar nombre de usuario en Auth0](https://deacero.atlassian.net/browse/DMP-498)

## How is done

1. Created a new type that defines the shopping cart item structure.
2. Modified the shoppingCartContext to work with an array of the new type.
3. Created a new Card component.
4. Created a Product component that will display the data of a product added to the cart.
5. Added a delete function to the delete button in the cart product to remove it from the context.
6. Added a card containing the price details of the products as well as the quantity. It updates when removing an item.

## How to test

1. Run the app `npm start`.
2. Go to `/carrito/productos`.
3. You will see that two products are pre-added to the cart (this is just for testing purposes).
4. Check that the subtotal card has the correct sum of prices, as well as the correct sum of products.
5. On any card, click on the `Eliminar` button, this will remove it from the list. Check that the "subtotal" section gets updated as well.
